src/middlewared/middlewared/alembic/versions/25.04/2024-09-19_18-48_user_api_key.py

```diff
@@ -0,0 +1,42 @@
+"""Convert to user-linked tokens
+
+Revision ID: 8ae49ac78d14
+Revises: 6dedf12c1035
+Create Date: 2024-09-19 18:48:55.972115+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import json
+
+
+# revision identifiers, used by Alembic.
+revision = '8ae49ac78d14'
+down_revision = '6dedf12c1035'
+branch_labels = None
+depends_on = None
+
+DEFAULT_ALLOW_LIST = [{"method": "*", "resource": "*"}]
+ENTRY_REVOKED = -1
+
+
+def upgrade():
+    conn = op.get_bind()
+    to_revoke = []
+    for row in conn.execute("SELECT id, allowlist FROM account_api_key").fetchall():
+        try:
+            if json.loads(row['allowlist']) != DEFAULT_ALLOW_LIST:
+                to_revoke.append(str(row['id']))
+        except Exception:
```

Comment:
Should we log it somewhere ?

Reply:
We don't do that in other migrations, but I think I should add an alert for revoked API keys.